src/blocker.rs

```diff
@@ -1623,6 +1573,175 @@ mod blocker_tests {
         }
     }
 
+    #[test]
+    fn test_removeparam() {
+        let filters = vec![
+            String::from("||example.com^$removeparam=test"),
```

Comment:
Should we also test the `mkt_tok` case where that param needs to be removed unless there is an `mkt_unsubscribe` param in the query string? (Or maybe that's not implemented yet? I couldn't find the syntax for it in https://kb.adguard.com/en/general/how-to-create-your-own-ad-filters#removeparam)

Reply:
We could do something like `mkt_unsubscribe$removeparam=mkt_tok` - that'd remove `mkt_tok` whenever `mkt_unsubscribe` appeared _anywhere_ in the URL (not just in the query string). The "anywhere" part is probably fine. But I understand the logic is the opposite of what we want, which would be more like:

```adblock
! remove mkt_tok everywhere
*$removeparam=mkt_tok
! *except* when there's an mkt_unsubscribe in the URL
@@mkt_unsubscribe$removeparam=mkt_tok
```

That's gonna require support for removeparam exceptions, which aren't in the uBO default lists and I figured would be best handled in a followup.

Reply:
> That'd remove mkt_tok whenever mkt_unsubscribe appeared anywhere in the URL (not just in the query string). The "anywhere" part is probably fine.

Yes, that's also what we settled on in https://github.com/brave/brave-core/pull/13726.

> That's gonna require support for removeparam exceptions, which aren't in the uBO default lists and I figured would be best handled in a followup.

Sounds good. No need to add `mkt_unsubscribe$removeparam=mkt_tok` since as you point out, it's the opposite of what we need anyways. We can add the real one later once exceptions are implemented.